cli/src/commands/git/fetch.rs

```diff
@@ -35,30 +35,43 @@ use crate::git_util::print_git_import_stats;
 use crate::git_util::with_remote_git_callbacks;
 use crate::ui::Ui;
 
+/// Specifies which branches to fetch from remotes
+#[derive(Clone, Debug)]
+enum BranchesToFetch<'a> {
+    /// Fetch specific branch patterns
+    Patterns(&'a [StringPattern]),
+    /// Fetch only tracked branches, optionally filtered by explicit patterns
```

Comment:
(The major thing we have to agree on) When I was suggesting `jj git fetch -b 'glob:ig/*' --tracked`, I wanted this to fetch all the tracked branches *in addition to* any `ig/*` branches.

I'm not sure whether fetching the intersection would be more intuitive to other people. I don't feel like it, but maybe I'm too swayed by my own use-case. There's also the matter that if you fetch the union, you'd still need to track the extra branches manually (though `jj branch track 'glob:ig/*'` is pretty intuitive).

If this is unclear, it might be easier to just disallow `--tracked -b` for now.

Reply:
I thought they would intersect, but in `jj git push`, some options are additive, and some combinations are banned. I agree it's better to disallow `--tracked -b`. We'll need a "branch expression" language.